pkg/proxy/nftables/helpers_test.go

```diff
@@ -815,3 +776,321 @@ func Test_diffNFTablesChain(t *testing.T) {
+		add element ip kube-proxy service-ips { 192.168.99.33 . tcp . 80 : goto external-LAUZTJTB-ns4/svc4/tcp/p80 }
+
+		# svc5
+		add set ip kube-proxy affinity-GTK6MW7G-ns5/svc5/tcp/p80__10.180.0.3/80 { type ipv4_addr ; flags dynamic,timeout ; timeout 10800s ; }
+		add rule ip kube-proxy service-HVFWP5L3-ns5/svc5/tcp/p80 ip daddr 172.30.0.45 tcp dport 80 ip saddr != 10.0.0.0/8 jump mark-for-masquerade
+		add rule ip kube-proxy service-HVFWP5L3-ns5/svc5/tcp/p80 ip saddr @affinity-GTK6MW7G-ns5/svc5/tcp/p80__10.180.0.3/80 goto endpoint-GTK6MW7G-ns5/svc5/tcp/p80__10.180.0.3/80
+		add rule ip kube-proxy service-HVFWP5L3-ns5/svc5/tcp/p80 numgen random mod 1 vmap { 0 : goto endpoint-GTK6MW7G-ns5/svc5/tcp/p80__10.180.0.3/80 }
+		add rule ip kube-proxy external-HVFWP5L3-ns5/svc5/tcp/p80 jump mark-for-masquerade
+		add rule ip kube-proxy external-HVFWP5L3-ns5/svc5/tcp/p80 goto service-HVFWP5L3-ns5/svc5/tcp/p80
+
+		add rule ip kube-proxy endpoint-GTK6MW7G-ns5/svc5/tcp/p80__10.180.0.3/80 ip saddr 10.180.0.3 jump mark-for-masquerade
+		add rule ip kube-proxy endpoint-GTK6MW7G-ns5/svc5/tcp/p80__10.180.0.3/80 update @affinity-GTK6MW7G-ns5/svc5/tcp/p80__10.180.0.3/80 { ip saddr }
+		add rule ip kube-proxy endpoint-GTK6MW7G-ns5/svc5/tcp/p80__10.180.0.3/80 meta l4proto tcp dnat to 10.180.0.3:80
+
+		add rule ip kube-proxy firewall-HVFWP5L3-ns5/svc5/tcp/p80 ip saddr != { 203.0.113.0/25 } drop
+
+		add element ip kube-proxy service-ips { 172.30.0.45 . tcp . 80 : goto service-HVFWP5L3-ns5/svc5/tcp/p80 }
+		add element ip kube-proxy service-ips { 5.6.7.8 . tcp . 80 : goto external-HVFWP5L3-ns5/svc5/tcp/p80 }
+		add element ip kube-proxy service-nodeports { tcp . 3002 : goto external-HVFWP5L3-ns5/svc5/tcp/p80 }
+		add element ip kube-proxy firewall-ips { 5.6.7.8 . tcp . 80 comment "ns5/svc5:p80" : goto firewall-HVFWP5L3-ns5/svc5/tcp/p80 }
+
+		# svc6
+		add element ip kube-proxy no-endpoint-services { 172.30.0.46 . tcp . 80 comment "ns6/svc6:p80" : goto reject-chain }
+		`)
+
+	nft := knftables.NewFake(knftables.IPv4Family, "kube-proxy")
+	err := nft.ParseDump(rules)
+	if err != nil {
+		t.Fatalf("failed to parse given nftables rules: %v", err)
+	}
+	// ensure rules were parsed correctly
+	assertNFTablesTransactionEqual(t, getLine(), rules, nft.Dump())
+	runPacketFlowTests(t, getLine(), nft, testNodeIPs, []packetFlowTest{
+		{
+			name:     "no match",
+			sourceIP: "10.0.0.2",
+			destIP:   "10.0.0.3",
+			destPort: 80,
+			output:   "",
+		},
+		{
+			name:     "single endpoint",
+			sourceIP: "10.0.0.2",
+			destIP:   "172.30.0.41",
+			destPort: 80,
+			output:   "10.180.0.1:80",
+		},
+		{
+			name:     "multiple endpoints",
+			sourceIP: "10.0.0.2",
+			destIP:   "172.30.0.44",
+			destPort: 80,
+			output:   "10.180.0.4:80, 10.180.0.5:80",
+		},
+		{
+			name:     "LOCAL, mark for masquerade",
+			sourceIP: testNodeIP,
+			destIP:   "192.168.99.22",
+			destPort: 80,
+			output:   "10.180.0.2:80",
+			masq:     true,
+		},
+		{
+			name:     "DROP",
```

Comment:
(It's fine to keep this in all-caps though since the tracePacket output is still "DROP" even though the nftables rule is "drop").